tests/Doctrine/Tests/Common/Proxy/NullableTypeHintsClass.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace Doctrine\Tests\Common\Proxy;
+
+/**
+ * Test PHP 7.1 nullable type hints / return types class.
+ */
+class NullableTypeHintsClass
+{
+    public function nullableTypeHintInt(?int $param)
+    {
+    }
+
+    public function nullableTypeHintObject(?\stdClass $param)
+    {
+    }
+
+    public function nullableTypeHintSelf(?self $param)
+    {
+    }
+
+    public function nullableTypeHintWithDefault(?int $param = 123)
+    {
+    }
+
+    public function nullableTypeHintWithDefaultNull(?int $param = null)
```

Comment:
There is still no way to detect this correctly at runtime (makes one test fail).